kolibri/core/content/utils/import_export_content.py

```diff
@@ -170,52 +210,160 @@ def get_import_export_data(  # noqa: C901
                 )
             )
 
-        count_content_ids = nodes_segment.count()
+        min_boundary += dynamic_chunksize
+
         # Only bother with this query if there were any resources returned above.
-        if count_content_ids:
-            number_of_resources = number_of_resources + count_content_ids
+
+        if nodes_query.count() > 0:
+            nodes_queries_list.append(nodes_query)
+
+    return nodes_queries_list
+
+
+def get_content_nodes_data(
+    channel_id, nodes_queries_list, available=None, topic_thumbnails=True
+):
+    """
+    Returns a set of resources, file names, and a total size in bytes for all
+    data files associated with the content nodes in nodes_queries_list.
+    """
+
+    queried_file_objects = {}
+    number_of_resources = 0
+
+    for nodes_query in nodes_queries_list:
+        number_of_resources = number_of_resources + nodes_query.count()
+
+        file_objects = LocalFile.objects.filter(
+            files__contentnode__in=nodes_query
+        ).values("id", "file_size", "extension")
+        if available is not None:
+            file_objects = file_objects.filter(available=available)
```

Comment:
:+1: although this could be merged with the previous `.filter()`, querysets are lazy so they can be chained like this and it's more readable.

Reply:
Yeah, this should be the same as the code from [`get_import_export_data`](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/utils/import_export_content.py#L160-L201) in the develop branch, except the indentation changed slightly :)